### PR TITLE
full flow GitHub to jira

### PR DESCRIPTION
**Pull request by ofersk**
a full test of the flow starting with GitHub and ending with Jira.

Note: the test is set to skip.

### CODE REVIEW

File: src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+        hookTargetURL = properties.getProperty("target");
+        COMMIT_FILE_PATH = String.format("%s/%s/%s/contents/%s", gitHubProperties.getApiUrl(), namespace, repo,
+                filePath);
+        REPO_HOOKS_BASE_URL = String.format("%s/%s/%s/hooks", gitHubProperties.getApiUrl(), namespace, repo);
+
+        CustomAsynchronousJiraRestClientFactory factory = new CustomAsynchronousJiraRestClientFactory();
+        try {
+            this.jiraURI = new URI(jiraProperties.getUrl());
+        } catch (URISyntaxException e) {
+            fail("Error constructing URI for JIRA");
```
Comment (natalieb07) [inline src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java]: Should we know some info from the exception?

File: src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        });
+        if (getStatus(COMMIT_FILE_PATH).equals(HttpStatus.OK)) {
+            deleteFile(COMMIT_FILE_PATH);
+        }
+        for (String issueCreatedKey : issueCreatedKeys) {
+            deleteIssue(issueCreatedKey);
+        }
+    }
+
+    @Given("source is Githb")
```
Comment (natalieb07) [inline src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java]: Should it be Gith**u**b? :)

File: src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+        JSONArray hooks = getJSONArray(REPO_HOOKS_BASE_URL);
+
+        if (!hooks.isEmpty()) {
+            fail("repository alredy has hooks configured");
+        }
+
+        String data = null;
+        try {
+            data = generateHookData(hookTargetURL, gitHubProperties.getWebhookToken());
+            // System.out.println(data);
```
Comment (natalieb07) [inline src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java]: need to remove the commented out line

File: src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+            ObjectNode jo = mapper.createObjectNode();
+            jo.put("message", "GitHubToJira test message");
+            ObjectNode committer = mapper.createObjectNode();
+            committer.put("name", "cxflowtestuser");
+            committer.put("email", "cxflowtestuser@checkmarx.com");
+            jo.set("committer", committer);
+            jo.put("content", content);
+
+            data = mapper.writeValueAsString(jo);
+            // System.out.println(data);
```
Comment (natalieb07) [inline src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java]: need to remove the commented out line

File: src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+        final HttpHeaders headers = getHeaders();
+        final HttpEntity<String> request = new HttpEntity<>(data, headers);
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(COMMIT_FILE_PATH, HttpMethod.PUT, request,
+                    String.class);
+            createdFileSha = new JSONObject(response.getBody()).getJSONObject("content").getString("sha");
+        } catch (Exception e) {
+            fail("faild to push a file: " + e.getMessage());
+        }
+        // System.out.println("pushed a new file");
```
Comment (natalieb07) [inline src/test/java/com/checkmarx/flow/cucumber/integration/end2end/github2jira/GitHubToJiraSteps.java]: same here

File: src/test/resources/cucumber/features/e2eTests/githubToJira.feature [hunk 6]
```diff
@@ -1,3 +1,6 @@
+@EndToEnd
+@Integration
```
Comment (natalieb07) [inline src/test/resources/cucumber/features/e2eTests/githubToJira.feature]: do we add both @EndToEnd and @Integration ?

**Comment by ofersk** [comment]
I rebased the code to be on top of the updated develop branch (the develop branch was broken, and now it passes the test before and after my code)

**Pull request merged**